### PR TITLE
Enable Past CI

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Enable Past CI

### CODE REVIEW

File: docker/transformers-past-gpu/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# Use installed torch version for `torch-scatter` to avid to deal with PYTORCH='pre'.
+# If torch is nightly version, the link is likely to be invalid, but the installation falls back to the latest torch-scatter
+### RUN python3 -m pip install --no-cache-dir torch-scatter -f https://data.pyg.org/whl/torch-$(python3 -c "from torch import version; print(version.__version__.split('+')[0])")+$CUDA.html
+### RUN python3 -m pip install --no-cache-dir intel_extension_for_pytorch==$INTEL_TORCH_EXT+cpu -f https://software.intel.com/ipex-whl-stable
+
+### RUN python3 -m pip install --no-cache-dir git+https://github.com/facebookresearch/detectron2.git pytesseract https://github.com/kpu/kenlm/archive/master.zip
+RUN python3 -m pip install -U "itsdangerous<2.1.0"
+
+### RUN python3 -m pip install --no-cache-dir git+https://github.com/huggingface/accelerate@main#egg=accelerate
```
Comment (ydshieh) [inline docker/transformers-past-gpu/Dockerfile:51]: no `accelerate`

File: docker/transformers-past-gpu/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# Install the target framework
+RUN echo "INSTALL_CMD = $INSTALL_CMD"
+RUN $INSTALL_CMD
+
+# Use installed torch version for `torch-scatter` to avid to deal with PYTORCH='pre'.
+# If torch is nightly version, the link is likely to be invalid, but the installation falls back to the latest torch-scatter
+### RUN python3 -m pip install --no-cache-dir torch-scatter -f https://data.pyg.org/whl/torch-$(python3 -c "from torch import version; print(version.__version__.split('+')[0])")+$CUDA.html
+### RUN python3 -m pip install --no-cache-dir intel_extension_for_pytorch==$INTEL_TORCH_EXT+cpu -f https://software.intel.com/ipex-whl-stable
+
+### RUN python3 -m pip install --no-cache-dir git+https://github.com/facebookresearch/detectron2.git pytesseract https://github.com/kpu/kenlm/archive/master.zip
```
Comment (ydshieh) [inline docker/transformers-past-gpu/Dockerfile:48]: not to include those 3rd party libs

File: utils/past_ci_versions.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import argparse
```
Comment (ydshieh) [inline utils/past_ci_versions.py:1]: This file is to make the installation commands easier to read in Docker files.
This should be used for other Docker files once this PR is merged.

For example, the following block hurts my head a bit already 😄 

https://github.com/huggingface/transformers/blob/49cd736a288a315d741e5c337790effa4c9fa689/docker/transformers-pytorch-gpu/Dockerfile#L19-L21

File: utils/past_ci_versions.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import os
+
+
+past_versions_testing = {
+    "pytorch": {
+        "1.10": {
+            "torch": "1.10.2",
+            "torchvision": "0.11.3",
+            "torchaudio": "0.10.2",
+            "python": 3.9,
```
Comment (ydshieh) [inline utils/past_ci_versions.py:11]: This is the highest python version for that PyTorch version. Not really important

File: utils/past_ci_versions.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        # need python 3.7
+        # "1.3": {
+        #     "torch": "1.3.1",
+        #     "torchvision": "0.4.2",
+        #     "torchaudio": None,
+        #     "python": 3.7,
+        #     "cuda": "cu100",
+        #     "docker-base": "10.0-cudnn7-devel-ubuntu18.04",
+        #     "install": "python3 -m pip install --no-cache-dir -U torch==1.3.1 torchvision==0.4.2 torchaudio==0.4.0 --extra-index-url https://download.pytorch.org/whl/cu100",
+        # },
```
Comment (LysandreJik) [inline utils/past_ci_versions.py:94]: Let's remove it then

File: utils/past_ci_versions.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    },
+    # Should we include `tensorflow_probability`?
```
Comment (LysandreJik) [inline utils/past_ci_versions.py:96]: Let's do without it as a first step, let's start simple

**Review by LysandreJik — APPROVED** [review]
This is great, super impressive! Let's merge it when you're happy with it and monitor failures. Please open an issue with the first reports when they're here so that we may discuss there which versions we'd like to keep or not.

File: utils/past_ci_versions.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        # need another `nvidia:cuda` docker image, otherwise GPU not working
+        "2.4": {
+            "tensorflow": "2.4.4",
+            "install": "python3 -m pip install --no-cache-dir -U tensorflow==2.4.4",
+            "base_docker": "nvidia/cuda:11.0.3-cudnn8-devel-ubuntu20.04",
+        },
+        # "2.3": {
+        #     "tensorflow": "2.3.4",
+        #     "install": "python3 -m pip install --no-cache-dir -U tensorflow==2.3.4",
+        # },
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply utils/past_ci_versions.py:123]: TF 2.3 requires CUDA 10.1 / cudnn 7.6, for which I don't find a good docker image on `nvidia/`.

(There is one for Ubuntu 18.04, but I remembered I have some trouble with it. I can re-run docker build to see what goes wrong)

[the remark this answers is not in the document]
Comment (ydshieh) [reply utils/past_ci_versions.py:123]: Hi @LysandreJik . I remember what's wrong 😄 

The best image I could find is `nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04`, which has Python 3.6. During build,  I got errors like
```
ERROR: No matching distribution found for accelerate>=0.10.0; extra == "dev"
```
Also, we have `python": "python>=3.7.0` in `setup.py`.

As we discussed shortly before, I could try to build an image from scratch with python >= 3.7 with the necessary cuda/cudnn. I didn't do that though but just tried to launch the Past-CI with those versions easier to deal with.

Should I make it work (for TF 2.3) before merging this PR?

[the remark this answers is not in the document]
Comment (LysandreJik) [reply utils/past_ci_versions.py:123]: Let's merge it like this and ask the TF crew if they think 2.3 should be supported before putting too much efforts into it :)

**Comment by ydshieh** [comment]
> merge it ... and monitor failures. 

I read your comment too quickly. So far the Past CI will be triggered only on pushing to `run-past-ci*` branches. I ran it ~ June 20 however, and I opened #18181 today.

I think we can run past CI monthly or even bimonthly. Please let me know if you have different opinion, @LysandreJik. Thanks.

**Pull request merged**